### DISCUSSION TITLE
Support 2-level subblock (elements-containers + elements-definitions as children of blocks)

### DISCUSSION THREAD

**Discussion by psyafter** [voidlabs/mosaico#626] [Ideas]
Nice to have option for subblocks - drug and drop subblock to block

**Comment by bwl21** [comment]
while playing with another editor which allows sublocks, I find that the newsletter author is likely to be confused by an arbitrary nesting of blocks. OTH the request sounds reasoable. Maybe it makes sense so specify a list of possible parent blocks for a nested block.

**Comment by psyafter** [comment]
May be not whole subblock but set of possible elements.
For example I want to drugndrop  image to text, or some another elements. 
Or it can be form element that I want place in another block. And so I want add fields to form...

If it will be supported - so Mosaico can be not just first email template editor but even Landing Pages editor as well

**Comment by bago** [comment]
I think "free nesting" is not something we'll ever provide, because it is against our "design".
Instead I'd like to support the "subblock", or better, a "block"+"element" layers where a block can declare "elements container".

The only editors I know doing this (campaign monitor editor and beefree) do this with fixed elements. This is also against our current architecture where everything is defined by the template, but I think we can find some good way to implement it.

My main issues are about UX/UI for the "draggable/droppable" elements (development is not easy, but I don't fear it if I have a clear UX+template language idea).
